src/main/java/org/zalando/nakadi/service/kpi/publisher/NakadiKpiPublisher.java

```diff
@@ -0,0 +1,32 @@
+package org.zalando.nakadi.service.kpi.publisher;
+
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.util.FeatureToggleService;
+
+import java.util.function.Supplier;
+
+@Component
+public class NakadiKpiPublisher {
+
+    private final FeatureToggleService featureToggleService;
+    private final EventsProcessor eventsProcessor;
+
+    @Autowired
+    protected NakadiKpiPublisher(final FeatureToggleService featureToggleService,
+                                 final EventsProcessor eventsProcessor) {
+        this.featureToggleService = featureToggleService;
+        this.eventsProcessor = eventsProcessor;
+    }
+
+    public void publish(final String etName, final Supplier<JSONObject> data) {
```

Comment:
can you please rename `data` parameter

Reply:
One suggestion would be `kpiEntry`. What do you think?

Reply:
Or `kpiEvent`

Reply:
Well, I actually suggested to rename it in order to reflect that it is a supplier, not the actual data